### ISSUE TITLE
Ubuntu Pro on Azure fails to auto enable livepatch

### ISSUE THREAD

**Issue by alesstimec** [canonical/ubuntu-advantage-client#1235]
Hi,

Starting an Ubuntu Pro 20.04 LTS instance on Azure results in a VM that fails to enable livepatch.

Seems that we fail to install the canonical-livepatch snap. "ua status" reports:
"Invalid command specified '/snap/bin/canonical-livepatch status'."
for livepatch.

See logs:
- syslog: https://pastebin.canonical.com/p/tXQs5jXFHY/
- ubuntu-advantage: https://pastebin.canonical.com/p/JJz5J9p5wF/
- cloud-init-output: https://pastebin.canonical.com/p/K8srKtXQyk/

However if i ssh into the VM and run "sudo snap install canonical-livepatch" it successfully installs the snap.

**Comment by lucasmoura** [comment]
Hi @alesstimec, thanks for reporting this issue.

Looking at the logs, it seems that snapd is still running a configuration step. This is indicated by the error:
```
Failed running command '/usr/bin/snap install canonical-livepatch' [exit(10)]. Message: error: too early for operation, device not yet seeded or device model not acknowledged
```

One thing that we may do is, during auto-attach verify if we are going to install any snaps, and if we are, run this wait command before that:
```
snap wait system seed.loaded
```

This is a hypothesis, since I was not able to reproduce this bug yet. I have launched 3 focal PRO images and none of them had this issue, which is expected, since the error seems to be a race condition with snapd.

I will figure out a way to proper test this and if that is the issue, I will open a PR to address it

**Comment by alesstimec** [comment]
@lucasmoura i was only able to reproduce the bug (reliably) when starting the VM via azure API.. When starting the VM via Azure Portal it worked..

**Comment by blackboxsw** [comment]
@alesstimec Thanks again for the info here. on AWS PRO images I see snap.seeded.sevice taking about 30 seconds on system boot.

TLDR: your version of ua-client is only {'user-agent': 'UA-Client/20.3', in that image. We have already fixed livepatch install to run 
`snap wait system seed.loaded` before trying to install canonical-livepatch snap  https://github.com/canonical/ubuntu-advantage-client/blob/master/uaclient/entitlements/livepatch.py#L103 as lucas suggested above.

On my AWS system I see seeded service took 30 seconds

```
systemd-analyze blame | grep seed
         32.583s snapd.seeded.service
            29ms systemd-random-seed.service
```

On your system it looks like seeding took about a minute (from your syslog) 

```
Nov  9 09:25:28 ubuntu-pro-test systemd[1]: Starting Wait until snapd is fully seeded...
...
Nov  9 09:26:27 ubuntu-pro-test systemd[1]: Finished Wait until snapd is fully seeded.
```

One thing that could be done to "support" this old version of ubuntu-advantage-tools is launching with user-data
```
#cloud-config
runcmd:
  - snap wait system seed.loaded
   - ua enable livepatch
```

**Comment by cmars** [comment]
@blackboxsw @patviafore-canonical Is there a newer Pro image we should be using that has a newer version of the client?

Looking at the code @alesstimec used to launch Pro in Azure, it appears we're choosing the latest image for Publisher "canonical", Offer "0001-com-ubuntu-pro-focal", SKU "pro-20_04-lts".

Trying to figure out if we've just latched on to an old image, or if we need to update our VM image in the marketplace to fix this issue?

**Comment by blackboxsw** [comment]
@alesstimec can you post the output of ``cloud-init query --format="{{platform}}:{{ds.meta_data.imds.compute.offer}}:{{ds.meta_data.imds.compute.version}}"``

**Comment by blackboxsw** [comment]
Cross reference bug created against cloud-images for Azure focal  PRO: https://bugs.launchpad.net/cloud-images/+bug/1903753

**Comment by lucasmoura** [comment]
We believe this issue has already been solved. Please open this issue again if that is not the case
